File: makeimage.cpp
#include <iostream>
#include <fstream>
#include "Drawing.h"

using namespace std;

int main(int argc, char* argv[]) {
    const char* fName = "out.bmp";
    ofstream f;
    f.open(fName, ios::out | ios::binary);

    unsigned int blockSize = 100;
    Drawing d;
    for (int i = 0; i < 100; i++) {
        d.create(blockSize);
    }

    f.write((const char*) d.getBuffer(), d.getBufferSize());

    f.close();

    cout << "Done!" << endl;
    return 0;
}

File: java/src/org/duldeb/makeimage/MakeImage.java
package org.duldeb.makeimage;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;

public class MakeImage {

	public static void main(String[] args) {
		Drawing d = new Drawing();

		String fName = "out.bmp";

		int blockSize = 100;
		//for (int i = 0; i < 1000; i++) {
			d.create(blockSize);
		//}

		FileOutputStream f;
		try {
			f = new FileOutputStream(fName);
			f.write(d.getBuffer());
			f.close();
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		} catch (IOException e) {
			e.printStackTrace();
		}
    
        System.out.println("Done!");
	}
}
